### PR TITLE
📝  [Typing sagas] First draft

### PR DESCRIPTION
**Pull request by fTrestour**

### CODE REVIEW

**Review by hugomartinet — APPROVED** [review]

File: content/posts/2020-02-19-sagas-typing/index.mdx [hunk 1]
```diff
@@ -0,0 +1,287 @@
+---
+title: How to type a saga
+author: Fabien Trestour
+date: 2020-02-19
+hero: ./images/header.png
+excerpt: Redux-saga is a very useful tool to have in a front-end stack. Typing them properly is not obvious though. Here's a guide on how to do it!
```
Comment (nicgirault) [inline content/posts/2020-02-19-sagas-typing/index.mdx]: In articles, your words weight more when they are factual. `a very useful tool` is not factual, it's an opinion. It's not a problem but it is important to be aware of this. Something like `is a widely used library that help to deal with side effects in a front-end app` is more factual.

Comment (nicgirault) [reply content/posts/2020-02-19-sagas-typing/index.mdx]: Maybe speak about typescript 3.7 with a link to the doc to explain that generators can now be typed

File: content/posts/2020-02-19-sagas-typing/index.mdx [hunk 2]
```diff
@@ -0,0 +1,287 @@
+---
+title: How to type a saga
```
Comment (nicgirault) [inline content/posts/2020-02-19-sagas-typing/index.mdx]: Think about SEO when writing your title. I would say that you need "typescript", "redux-saga" in the words of the title

File: content/posts/2020-02-19-sagas-typing/index.mdx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+author: Fabien Trestour
+date: 2020-02-19
+hero: ./images/header.png
+excerpt: Redux-saga is a very useful tool to have in a front-end stack. Typing them properly is not obvious though. Here's a guide on how to do it!
+---
+
```
Comment (nicgirault) [inline content/posts/2020-02-19-sagas-typing/index.mdx:8]: I would copy the excerpt at the beginning of the article

File: content/posts/2020-02-19-sagas-typing/index.mdx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+```ts
+function* hello(foo: string) {}
+// -> (foo: string) => Generator
+```
+
+Even if `redux-saga` relies on other concepts (like effects), typing a saga mostly boils down to typing a generator.
+
+# Typing a generator
+
+Generators typing was improved in [TS 3.6](https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#stricter-generators).
```
Comment (nicgirault) [inline content/posts/2020-02-19-sagas-typing/index.mdx:24]: 👍

File: content/posts/2020-02-19-sagas-typing/index.mdx [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+  const bar: string = yield "baz";
+  // Type 'boolean' is not assignable to type 'string'.
+  console.log(bar);
+  return 0;
+}
+```
+
+# Typing a saga
+
+Everything up to this point works for any kind of generator factory. However, `redux-sagas` is not limited to writing generator factories.
```
Comment (nicgirault) [inline content/posts/2020-02-19-sagas-typing/index.mdx]: redux-saga

**Pull request merged**